clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -1101,6 +1101,37 @@ def BitfieldInfoAttr : CIR_Attr<"BitfieldInfo", "bitfield_info"> {
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// AnnotationAttr
+//===----------------------------------------------------------------------===//
+def AnnotationAttr : CIR_Attr<"Annotation", "annotation"> {
+  let summary = "Annotation attribute for global variables and functions";
+  let description = [{
+    Represent C/C++ attribute of annotate in CIR. example c code is:
+    ```
+     in test.c line 10
+     int *a __attribute__((annotate("testptr", "21", 12 ))); 
+    ```
+     this example code, the AnnotationAttr records annotation name "testptr", 
+     source file "test.c", line number 12, and arguments "21" and 12 for 
+     global variable a
+  }];
+
+  // Use ConstArrayAttr as type of name and translation unit
+  // because they will be used to generate global variable. Using ConstArrayAttr
+  // is a lot more suitable for that than StringAttr.
```

Comment:
Can you elaborate on why `ConstArrayAttr` is better than `StringAttr`? `StringAttr` seems like the way to go, did you see examples with `ConstArrayAttr` elsewhere?

Reply:
Both "testptr", "21" should be StringAttr

Reply:
The problem with StringAttr is that its not typed. My original attempt was using StringAttr, however, when I tried to create global string var during loweringPrepare, we need create a string type. Plus, we have to convert this StringAttr to ConstArray Str anyway to make it suitable as a init_value for the global string var created ( as init_value has to be typed). So, why not use ConstArray in the first place?

Reply:
We already discussed this, please let's not rehash. If you create a specific attribute like I suggested, e.g. `GlobalAnnotationAttr`, that can be typed to whatever we want the global type to be. The attribute should hold a ArrayAttr and it could also have an associated type, there are many examples.

Reply:
> So, why not use ConstArray in the first place?

This is bloating CIR and using C/C++ specific types for no good reason or usecase. It's complicating something that should be simple and direct.